woocommerce/woocommerce
# Add @woocommerce/create-product-editor-block package

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR introduces a new `@woocommerce/create-product-editor-block` package, which is a template that can be used with the `@wordpress/create-block` tool to quickly get the skeleton of a new block-based extension to the new block-based product editor up-and-running.

The skeleton project generated has support for:

- `wp-env`
    - WordPress 6.2
    - WooCommerce trunk nightly
    - WooCommerce Beta Tester
- linting and code formatting
- TypeScript

The example block generated simply renders a message. Support for interactivity in the example block will be added in future PRs.

<img width="1276" alt="Screenshot 2023-05-15 at 16 43 57" src="https://github.com/woocommerce/woocommerce/assets/2098816/f0dcefbe-d663-424c-b34e-87701b25fd54">


The aim is to get devs up and running with block-based extensibility in a few simple commands.

Closes #38143.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Follow the instructions in the package's README.md (note, since this package hasn't been published yet, you will need to use the `npx @wordpress/create-block --template ./path/to/woocommerce/packages/js/create-product-editor-block my-extension-name` form of the command, with a local path to the template)
2. Verify that the generated skeleton project can be used, with the built-in `wp-env` support, to render the example block in the block-based product form, with no modifications of any files or any commands other than those mentioned in the README.md.
3. Verify the generated skeleton project's files look reasonable.
